src/node_crypto.cc

```diff
@@ -5616,13 +5616,13 @@ void RandomBytesProcessSync(Environment* env,
 void RandomBytes(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
-  if (!args[0]->IsUint32()) {
+  if (!args[0]->IsNumber() || args[0]->NumberValue() < 0) {
```

Comment:
I wonder if the `args[0]->NumberValue()` has a performance impact here? Could you not use `args[0].As<Number>`?

Reply:
@evanlucas did you accidentally add a `not` in? Would you like me to change it to `as<Number>`? 

for the record `NumberValue` does   `if (obj->IsNumber()) return Just(obj->Number());`

Reply:
No, I thought it had come up a few times that .As() is faster than *Value(), but could be wrong. I haven't done any performance testing on the difference though. Being that this is used for things like uuid generation, checking which is faster would probably be a good thing though.

Reply:
`NumberValue()` has a fast path for things that are already numbers, so it won’t be a huge different. In any case, not compared to the work of actually creating these numbers.

The overload of `NumberValue()` that doesn’t take a `context` argument is deprecated, by the way; you should use the one that does. (The reason being that `NumberValue` can actually execute arbitrary code via `@@toPromitive` if the input isn’t already a number, but in this case that won’t happen).

I personally would really prefer `args[0].As<Number>` is you’re already checking that `args[0]` is a `Number` instance.